### PR TITLE
Speed-up server start-up by pre-fetching

### PR DESCRIPTION
**Pull request by Mykhailenko**
During start-up, server executes _loadJobsTasks_ query. And after go through all the retrieved tasks (N tasks) and for each task, access its collections as getEnvModifiers(), getSelectionScripts(), etc. Accessing these collection make Hibernate to send a request for each access. Thus for N tasks in DB, we will send 1+4N requests (so-called **hibernate n+1 problem**),  just to get all jobs and tasks. 

Solution is to fetch tasks, with its collections in single query. However, to fetch 2 collections of List type in one query, we have to specify "@OrderColumn" annotation, and of course, add additional column to SelectionScriptData and SelectorData. 
Of course, adding a non-nullable column will require special effort during platform migration.

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java [hunk 1]
```diff
@@ -45,12 +45,18 @@
...
 import javax.persistence.OneToMany;
 import javax.persistence.OneToOne;
+import javax.persistence.OrderColumn;
 import javax.persistence.Table;
 import javax.persistence.Transient;
 
 import org.hibernate.annotations.BatchSize;
 import org.hibernate.annotations.Cascade;
 import org.hibernate.annotations.CascadeType;
+import org.hibernate.annotations.Fetch;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java]: MINOR Remove this unused import 'org.hibernate.annotations.Fetch'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java [hunk 2]
```diff
@@ -45,12 +45,18 @@
...
+import org.hibernate.annotations.FetchMode;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java]: MINOR Remove this unused import 'org.hibernate.annotations.FetchMode'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java [hunk 3]
```diff
@@ -45,12 +45,18 @@
...
+import org.hibernate.annotations.IndexColumn;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java]: MINOR Remove this unused import 'org.hibernate.annotations.IndexColumn'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java [hunk 4]
```diff
@@ -45,12 +45,18 @@
...
+import org.hibernate.annotations.LazyCollection;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java]: MINOR Remove this unused import 'org.hibernate.annotations.LazyCollection'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java [hunk 5]
```diff
@@ -45,12 +45,18 @@
...
+import org.hibernate.annotations.LazyCollectionOption;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java]: MINOR Remove this unused import 'org.hibernate.annotations.LazyCollectionOption'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Mykhailenko** [comment]
Retest this please

**Pull request merged**